src/server/editorServices.ts

```diff
@@ -756,6 +756,9 @@ namespace ts.server {
         /*@internal*/
         readonly watchFactory: WatchFactory<WatchType, Project>;
 
+        /*@internal*/
+        private sharedExtendedConfigFilesMap = new Map<string, SharedExtendedConfigFileWatcher<Project>>();

```

Comment:
It would be better to have MultiMap<Path, Project>

Reply:
The key is canonical path for the extended config file

Reply:
Switched to `MultiMap<Path, ConfiguredProject>` for the config-project mappings and created another `Map<Path, FileWatcher>` to store the `FileWatcher` for each config.